app/models/ems_refresh/parsers/azure.rb

```diff
@@ -0,0 +1,214 @@
+module EmsRefresh
+  module Parsers
+    class Azure < ManageIQ::Providers::CloudManager::RefreshParser
+      VALID_LOCATION = /\w+/i
+
+      def self.ems_inv_to_hashes(ems, options = nil)
+        new(ems, options).ems_inv_to_hashes
+      end
+
+      def initialize(ems, options = nil)
+        @ems        = ems
+        ems.connect
+        @vmm        = ::Azure::Armrest::VirtualMachineManager.new
+        @asm        = ::Azure::Armrest::AvailabilitySetManager.new
+        @options    = options || {}
+        @data       = {}
+        @data_index = {}
+      end
+
+      def ems_inv_to_hashes
+        log_header = "MIQ(#{self.class.name}.#{__method__}) Collecting data for EMS : [#{@ems.name}] id: [#{@ems.id}]"
+
+        $log.info("#{log_header}...")
+        get_series
+        get_availability_sets
+        get_vms
+        clean_up_extra_flavor_keys
+        $log.info("#{log_header}...Complete")
+
+        @data
+      end
+
+      private
+
+      def get_series
+        series = []
+        get_locations.each do |location|
+          begin
+            series << @vmm.series(location)
+          rescue RestClient::BadGateway
```

Comment:
@blomquisg
We gather a list of locations, then for each location we request the series (flavors) that apply to that location.
For some unknown reason, when we request the series for some locations, such as AustraliaEast, it bombs out with a BadGateway exception. I could not figure out what the root cause of the problem was.